fhoeben/hsac-fitnesse-fixtures
# Json javadoc for contexthelp

Have javadoc json available in the jar's meta-inf, so autocompleteresponder can serve all relevant info to bootstrap-plus.

See https://gitlab.com/praegus/toolchain-fixtures/fixture-doc-generator-doclet for source and readme

To test:
Add the following execution to your build's maven dependency plugin:

```
 <execution>
	<id>copy-plugins</id>
	<phase>generate-resources</phase>
	<goals>
		<goal>copy</goal>
	</goals>
	<configuration>
		<artifactItems>
			<artifactItem>
				<groupId>nl.praegus</groupId>
				<artifactId>toolchain-fitnesse-plugin</artifactId>
				<version>1.15</version>
				<classifier>jar-with-dependencies</classifier>
				<overWrite>true</overWrite>
			</artifactItem>
		</artifactItems>
		<outputDirectory>${project.basedir}/wiki/plugins</outputDirectory>
	</configuration>
</execution>
```
To be able to use the plugins' 'debug script' table, make sure that `nl.praegus:toolchain-fixtures:1.14` is added as a dependency, `nl.praegus.fitnesse.slim.fixtures.util` is imported in slim and `pause test fixture` is added as a library.

Don't forget  to register bootstrap-plus as the theme (this will no longer be necessary from the next fitnesse and toolchain-plugin releases)

Now when editing a test, your context helper (top right) will contain all javadoc as documentation, as well as the names of all expected parameters, public constructors. These will also be available in codeMirror's autoComplete list and bootstrap-plus' validator.

Special pages or new test pages will get the context of the last test that was opened.
